roboteam_ai/src/control/ballHandling/BallHandlePosControl.cpp

```diff
@@ -91,58 +110,61 @@ RobotCommand BallHandlePosControl::getRobotCommand(const RobotPtr &r, const Vect
     bool ballIsMovingTooFast = ball->vel.length2() > minVelForMovingball*minVelForMovingball;
+    bool shouldGetBall = robotDoesNotHaveBall && (robotIsTooFarFromBall || ballIsMovingTooFast);
 
-    if (robotDoesNotHaveBall && (robotIsTooFarFromBall || ballIsMovingTooFast)) {
-        return goToBall(ballIsFarFromTarget);
+    if (ballIsOutsideField) {
+        status = HANDLING_BALL;
+        Vector2 targetBallPos = ControlUtils::projectPositionToWithinField(ball->pos, 1.0);
+        return handleBall(targetBallPos, BACKWARDS, shouldGetBall);
     }
 
-    // check if we are doing something already
-    if (dribbleBackwards->getBackwardsProgression() != DribbleBackwards::START) {
-        return dribbleBackwards->getRobotCommand(robot, targetPos, targetAngle);
-    }
+    if (preferredTravelStrategy == NO_PREFERENCE) {
+        if ((ballIsFarFromTarget && dribbleBackwards->getBackwardsProgression() !=
+            DribbleBackwards::BackwardsProgress::START)) {
 
-    if (dribbleForwards->getForwardsProgression() != DribbleForwards::START) {
-        return dribbleForwards->getRobotCommand(robot, targetPos, targetAngle);
+            dribbleBackwards->reset();
+        }
     }
 
-    switch (preferredTravelStrategy) {
-    case forwards: return dribbleForwards->getRobotCommand(robot, targetPos, targetAngle);
-    case backwards: return dribbleBackwards->getRobotCommand(robot, targetPos, targetAngle);
+    return handleBall(targetPos, preferredTravelStrategy, shouldGetBall, ballIsFarFromTarget);
+
+}
+
+void BallHandlePosControl::printStatus() {
+    switch (status) {
+    case GET_BALL:std::cout << "status: GET_BALL" << std::endl;return;
+    case HANDLING_BALL:std::cout << "status: HANDLING_BALL" << std::endl;return;
+    case FINALIZING:std::cout << "status: FINALIZING" << std::endl;return;
+    case SUCCESS:std::cout << "status: SUCCESS" << std::endl;return;
     default:
-    case no_preference: {
-        // choose based on distance from the ball to the target
-        if (ballIsFarFromTarget) {
-            return dribbleForwards->getRobotCommand(robot, targetPos, targetAngle);
-        }
-        return dribbleBackwards->getRobotCommand(robot, targetPos, targetAngle);
-    }
+    case FAILURE:std::cout << "status: FAILURE" << std::endl;return;
     }
-
 }
 
-RobotCommand BallHandlePosControl::goToBall(bool ballIsFarFromTarget) {
+RobotCommand BallHandlePosControl::goToBall(const Vector2 &targetBallPos, TravelStrategy travelStrategy, bool ballIsFarFromTarget) {
 
-    if (Constants::SHOW_BALL_HANDLE_DEBUG_INFO()) {
+    if (Constants::SHOW_FULL_BALL_HANDLE_DEBUG_INFO()) {
         std::cout << "we do not have a ball yet" << std::endl;
     }
 
     if (ball->vel.length2() > minVelForMovingball*minVelForMovingball) {
-        ball->pos += (ball->vel*0.5 + ball->vel.stretchToLength(pow(ball->vel.length2(), 1.0/8.0)));
+        ball->pos += (ball->vel*0.5 + ball->vel.stretchToLength(pow(ball->vel.length2(), 1.0/16.0)));
```

Comment:
This is still black magic. Can you not update the ballPosition but call it something else to atleast make sure if you read the 100 lines below this that you know this is NOT the original ball position?